dotnet/src/Backends.HuggingFace/HuggingFaceBackend.cs

```diff
@@ -0,0 +1,176 @@
+    private readonly HttpClientHandler _httpClientHandler;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="HuggingFaceBackend"/> class.
+    /// </summary>
+    /// <param name="baseUri">Base URI to call for backend operations.</param>
+    /// <param name="model">Model to use for backend operations.</param>
+    /// <param name="httpClientHandler">Instance of <see cref="HttpClientHandler"/> to setup specific scenarios.</param>
+    public HuggingFaceBackend(Uri baseUri, string model, HttpClientHandler httpClientHandler)
+    {
+        this._model = model;
+
+        this._httpClientHandler = httpClientHandler;
+        this._httpClient = new(this._httpClientHandler);
+
+        this._httpClient.BaseAddress = baseUri;
+        this._httpClient.DefaultRequestHeaders.Add("User-Agent", HttpUserAgent);
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="HuggingFaceBackend"/> class.
+    /// </summary>
+    /// <param name="baseUri">Base URI to call for backend operations in <see cref="string"/> format.</param>
+    /// <param name="model">Model to use for backend operations.</param>
+    /// <param name="httpClientHandler">Instance of <see cref="HttpClientHandler"/> to setup specific scenarios.</param>
+    public HuggingFaceBackend(string baseUri, string model, HttpClientHandler httpClientHandler)
+        : this(new Uri(baseUri), model, httpClientHandler)
+    {
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="HuggingFaceBackend"/> class.
+    /// Using default <see cref="HttpClientHandler"/> implementation.
+    /// </summary>
+    /// <param name="baseUri">Base URI to call for backend operations.</param>
+    /// <param name="model">Model to use for backend operations.</param>
+    public HuggingFaceBackend(Uri baseUri, string model)
+        : this(baseUri, model, new HttpClientHandler())
+    {
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="HuggingFaceBackend"/> class.
+    /// Using default <see cref="HttpClientHandler"/> implementation.
+    /// </summary>
+    /// <param name="baseUri">Base URI to call for backend operations in <see cref="string"/> format.</param>
+    /// <param name="model">Model to use for backend operations.</param>
+    public HuggingFaceBackend(string baseUri, string model)
+        : this(new Uri(baseUri), model)
+    {
+    }
+
+    /// <inheritdoc/>
+    public async Task<string> CompleteAsync(string text, CompleteRequestSettings requestSettings)
+    {
+        return await this.ExecuteCompleteRequestAsync(text);
+    }
+
+    /// <inheritdoc/>
+    public async Task<IList<Embedding<float>>> GenerateEmbeddingsAsync(IList<string> data)
+    {
+        return await this.ExecuteEmbeddingRequestAsync(data);
+    }
+
+    /// <inheritdoc/>
+    public void Dispose()
+    {
+        this._httpClient.Dispose();
+        this._httpClientHandler.Dispose();
```

Comment:
It's good practice not to dispose a resource you don't own/created because it can be used other code.

Reply:
Good catch, resolved.